WordPress/Classes/ViewRelated/Notifications/Comment Moderation/CommentModerationViewModel.swift

```diff
@@ -20,9 +35,22 @@ final class CommentModerationViewModel: ObservableObject {
     }
 
     func didTapPrimaryCTA() {
+        // TODO
         switch state {
         case .pending:
-            state = .approved(liked: false)
+//            isNotificationComment ? WPAppAnalytics.track(.notificationsCommentApproved,
+//                                                         withProperties: Constants.notificationDetailSource,
+//                                                         withBlogID: notification?.metaSiteID) :
+//                                    CommentAnalytics.trackCommentApproved(comment: comment)
+//
+            commentService.approve(comment, success: { [weak self] in
+                self?.state = .approved(liked: false)
+//                self?.showActionableNotice(title: ModerationMessages.approveSuccess)
+//                self?.refreshData()
+            }, failure: { /* [weak self] */ error in
+//                self?.displayNotice(title: ModerationMessages.approveFail)
+//                self?.commentStatus = CommentStatusType.typeForStatus(self?.comment.status)
+            })
```

Comment:
I left these as a reminder to not miss the existing functionality. Will be removed once the actions are implemented.